contributors/onboarding/staff-onboarding-guide/engineer-onboarding.md

```diff
@@ -0,0 +1,111 @@
+
+## Week 1: Focus on environment setup and introductions
+
+* Set up equipment and environment (e.g. complete the day 1 checklist)
+  * Get laptop, set up developer environment
+  * Clone enterprise repo
+  * Log in to all accounts (Gmail, GitHub, JIRA, OneLogin, Community, Community Daily)
+  * Expense equipment
+  * HR set up (benefits, health insurance, etc.)
+* Meet the team
+  * Meet with a different team member for 30min each day
+* Meet with your dev lead
+  * Meet for at least 10-15 minutes everyday
+* Attend weekly team/company meetings to get the feeling for them
+  * Sprint Planning
+  * Company All Hands
+  * Developer Meeting
+  * R&D Meeting
+  * Developer Hangouts
+  * Video Game Hour
+* Join important channels
+  * https://community.mattermost.com/core/channels/developers
+  * https://community.mattermost.com/core/channels/developers-meeting
+  * https://community.mattermost.com/core/channels/thank-you
+  * https://community.mattermost.com/private-core/channels/announcements
+  * https://community.mattermost.com/private-core/channels/confidential-bugs
+  * https://community.mattermost.com/private-core/channels/off-topic
+* Read and understand the core values
+  * https://handbook.mattermost.com/company/about-mattermost#leadership-principles
+* Review video calling etiquette
+  * http://hulen.com/video-conf-peevs/
+* Work on 1-3 small tickets to get used to dev processes
+* Introduction at the R&D Meeting
+
+## Week 2: Focus on digesting information dump
+
+* Continue getting used to processes and workflows
+  * Sit in on some spec/design/technical meetings, your dev lead should send you invites to these meetings
+  * Do some code reviews, your dev lead should assign you to some
+* Introduction meetings
+  * Meet with a different team member for 30min each day
+* Mentor for the week is a senior teammate
+  * Meet for at least 10-15 minutes everyday
+* Meet with dev lead
+  * Meet 2-3 times in the week
+  * Go over core values with dev lead
+* Work on 1-3 small tickets
+* Introduction at the [Customer Obsession Meeting](https://handbook.mattermost.com/operations/operations/company-cadence#customer-obsession-meeting-aka-com)
+
+## Weeks 3-4: Focus on solidifying role in the team
+
+* Start taking a more active role as a member of your dev team
+* Begin to participate in some spec/design/technical meetings
+  * Provide feedback/ask questions, have an active role in the meetings
+* Mentor for each week is a different teammate
+  * Meet for at least 10-15 minutes everyday
+* Meet with dev lead
+  * Meet once a week and start regular weekly 1-1s
+* Work on some medium size tickets
+
+## Weeks 5-8 (Month 2): Work on your first project as dev owner
+
+* Begin work on your first large ish task and be the dev owner of it
```

Comment:
```suggestion
* Begin work on your first relatively large task and be the dev owner of it
```
Can we qualify "relatively large"? Maybe we should add the mana expectation?

Reply:
^ @lieut-data this would be for you (or another dev lead) to confirm

Reply:
I'm fine with `relative large` task, and might define that as 40 mana.